src/components/modules/toolbar/blockSettings.ts

```diff
@@ -197,6 +198,102 @@ export default class BlockSettings extends Module<BlockSettingsNodes> {
     }
   };
 
+  /**
+   * Returns list of items to be displayed in block tunes menu.
+   * Merges tool specific tunes, conversion menu and common tunes in one list in predefined order
+   *
+   * @param toolName - current block tool name
+   * @param commonTunes – common tunes
+   * @param toolTunes - tool specific tunes
+   */
+  private getTunesItems(toolName: string, commonTunes: TunesMenuConfigItem[], toolTunes?: TunesMenuConfigItem[]): PopoverItemParams[] {
+    const items = [] as TunesMenuConfigItem[];
+
+    if (toolTunes !== undefined && toolTunes.length > 0) {
+      items.push(...toolTunes);
+      items.push({
+        type: 'separator',
+      });
+    }
+
+    const convertToItems = this.getConvertToItems(toolName);
+
+    if (convertToItems.length > 0) {
+      items.push({
+        type: 'default',
+        icon: IconReplace,
+        title: I18n.ui(I18nInternalNS.ui.popover, 'Convert to'),
+        children: {
+          items: convertToItems,
+        },
+      });
+      items.push({
+        type: 'separator',
+      });
+    }
+
+    items.push(...commonTunes);
+
+    return items.map(tune => this.resolveTuneAliases(tune));
+  }
+
+  /**
+   * Returns list of conversion menu items available for the current block
+   *
+   * @param currentToolName - current block tool name
+   */
+  private getConvertToItems(currentToolName: string): TunesMenuConfigItem[] {
```

Comment:
Looks like the result could be cached somewhere

Reply:
Good point, fixed